contracts/ThreeHeapOrdering.sol

```diff
@@ -148,83 +150,90 @@ library ThreeHeapOrdering {
     /// @dev Only call this function when `_id` is not in the `_heap`.
     /// @dev Reverts with AddressIsZero if `_value` is 0.
     /// @param _heap The heap to modify.
+    /// @param _size The computed size of the heap.
     /// @param _id The address of the account to insert.
     /// @param _value The value of the account to insert.
     /// @param _maxSortedUsers The maximum size of the heap.
     function insert(
         HeapArray storage _heap,
+        uint256 _size,
         address _id,
         uint96 _value,
         uint256 _maxSortedUsers
     ) private {
         // `_heap` cannot contain the 0 address.
         if (_id == address(0)) revert AddressIsZero();
 
-        uint256 size = _heap.size;
         uint256 accountsLength = _heap.accounts.length;
 
         _heap.accounts.push();
 
-        if (size != accountsLength) setAccount(_heap, _heap.accounts[size], accountsLength);
+        if (_size != accountsLength) setAccount(_heap, _heap.accounts[_size], accountsLength);
 
-        shiftUp(_heap, Account(_id, _value), size);
-        _heap.size = computeSize(size + 1, _maxSortedUsers);
+        shiftUp(_heap, Account(_id, _value), _size);
+        _heap.size = computeSize(_size + 1, _maxSortedUsers);
     }
 
     /// @notice Decreases the amount of an account in the `_heap`.
     /// @dev Only call this function when `_id` is in the `_heap` with a value greater than `_newValue`.
     /// @param _heap The heap to modify.
+    /// @param _size The computed size of the heap.
     /// @param _id The address of the account to decrease the amount.
     /// @param _newValue The new value of the account.
     function decrease(
         HeapArray storage _heap,
+        uint256 _size,
         address _id,
         uint96 _newValue
     ) private {
         uint256 index = _heap.indexOf[_id];
 
-        if (index < (_heap.size - 1) / 3) shiftDown(_heap, Account(_id, _newValue), index);
+        // We only need to take care of sorting if there are nodes below in the heap.
+        if (3 * index + 1 < _size) shiftDown(_heap, _size, Account(_id, _newValue), index);
         else _heap.accounts[index].value = _newValue;
     }
 
     /// @notice Increases the amount of an account in the `_heap`.
     /// @dev Only call this function when `_id` is in the `_heap` with a smaller value than `_newValue`.
     /// @param _heap The heap to modify.
+    /// @param _size The computed size of the heap.
```

Comment:
I would put "the size of the heap" everywhere

Reply:
Computed is nice because it reminds the reader of the `computeSize` function. This is not the size that is stored in the DLL so I think it is worth adding this to differentiate

Reply:
I think that for `increase` at least, the size is the one stored in the heap

Reply:
And I think that it is always the case

Reply:
> I think that for increase at least, the size is the one stored in the heap

No it's not the one stored in the heap because `_maxSortedUser` can change from transaction to transaction

Reply:
But when we call update we store the new size right ?

Reply:
Oh ok I see what you mean. I was thinking about the size before the transaction. Adding computed insists on the fact that it is not the size stored in the heap before the transaction, so I feel like it's worth it to add it. If you are not convinced we can go back, it's also ok with me

Reply:
Ok I see. Then can you add it everywhere please ?

Reply:
Otherwise replace by « new »

Reply:
Ok done. I kept computed because new can be misleading in the case of insert

Reply:
Ok perfect